constraint/pkg/client/drivers/local/local.go

```diff
@@ -574,9 +583,50 @@ func (d *Driver) RemoveTemplate(ctx context.Context, templ *templates.Constraint
 	kind := templ.Spec.CRD.Spec.Names.Kind
 	namePrefix := createTemplatePath(targetHandler, kind)
 	_, err := d.deleteModules(namePrefix)
+	delete(d.templateHandler, templ.Spec.CRD.Spec.Names.Kind)
```

Comment:
This `delete` needs to be protected by a mutex since it is possible multiple callers could call `RemoveTemplate` simultaneously. It should be fine to use `d.modulesMux`

Reply:
Ditto other reads/writes for d.TemplateHandler.

Alternatively, you could use a sync.Map if you don't want to worry about this. While less performant than RWMutex, it's easier to read.

Reply:
Also feel free to just make a RWMutex just for d.templateHandler - any of these options should work.

Reply:
What happens if a template that has pre-existing constraints is deleted?

Reply:
Right now it is Client's responsibility to coordinate cascading deletes. So this doesn't change that flow. As a future PR, cascading deletes should be handled by Drivers.